### PR TITLE
Fix patch for getAttachedShaders

### PR DESCRIPTION
**Pull request by ShivanKaul**
Resolves https://github.com/brave/brave-browser/issues/37044

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Go to https://shivankaul.com/brave/webgl/get_attached_shaders.html
You should see: `Attached shaders: [object WebGLShader]` printed out on the page.

### CODE REVIEW

File: patches/third_party-blink-renderer-modules-webgl-webgl_rendering_context_base.cc.patch [hunk 1]
```diff
@@ -22,8 +22,8 @@ index 9aab2ce4deeacb064d1b6048eb930b24354f9858..df329d23c618e8e6060664d3a37c85b6
  std::optional<HeapVector<Member<WebGLShader>>>
  WebGLRenderingContextBase::getAttachedShaders(WebGLProgram* program) {
    if (!ValidateWebGLProgramOrShader("getAttachedShaders", program))
-+  BRAVE_WEBGL_RENDERING_CONTEXT_BASE_NULLOPT
      return std::nullopt;
++  BRAVE_WEBGL_RENDERING_CONTEXT_BASE_NULLOPT
```
[the remark this answers is not in the document]
Comment (bridiver) [reply patches/third_party-blink-renderer-modules-webgl-webgl_rendering_context_base.cc.patch:26]: nvm, I thought `ValidateWebGLProgramOrShader` was defined somewhere else, but we could do this in `WebGL2RenderingContextBase` with `isContextLost` and `kFunctionName`

[the remark this answers is not in the document]
Comment (bridiver) [reply patches/third_party-blink-renderer-modules-webgl-webgl_rendering_context_base.cc.patch:26]: actually we could do one patch in `ValidateWebGLProgramOrShader` to accomplish the same thing and remove the rest of the patches

[the remark this answers is not in the document]
Comment (bridiver) [reply patches/third_party-blink-renderer-modules-webgl-webgl_rendering_context_base.cc.patch:26]: maybe `AllowFingerprintingForFunctionNameAndHost` so it doesn't affect other calls to `AllowFingerprintingForHost`

[the remark this answers is not in the document]
Comment (ShivanKaul) [reply patches/third_party-blink-renderer-modules-webgl-webgl_rendering_context_base.cc.patch:26]: https://github.com/brave/brave-browser/issues/39524 for follow-ups

**Comment by bsclifton** [comment]
Super late to the game 😅 but compiled locally and verified this works great

**Comment by kjozwiak** [comment]
Using `1.69.62 Chromium: 127.0.6533.17` and the STR/Cases outlined via https://github.com/brave/brave-core/pull/24446#issue-2384800053 & https://github.com/brave/brave-browser/issues/37044#issue-2205323588, reproduced the original issue as per the following:

**`Example`** | **`Example`**
----------------|-----------------
image | image

**Pull request merged**